source/planner.js

```diff
@@ -10,7 +10,185 @@ function init(){
     const slider = document.getElementById('importance-slider');
     const switchToMonthlyButton = document.getElementById('switch-to-monthly');
     const switchToWeeklyButton = document.getElementById('switch-to-weekly');
+    
+    //Add Task Button//
+    const addTaskButton = document.getElementById('add-task-button');
+    const addTaskPopup = document.getElementById('add-task-popup');
+    const addTaskConfirm = document.getElementById('add-task-confirm');
+    const addTaskCancel = document.getElementById('add-task-cancel');
+
+    // Insert the HTML structure for the add task popup
+    document.body.insertAdjacentHTML('beforeend', `
+    <div class="popup" id="add-task-popup">
+        <h2>Add New Task</h2>
+        <input type="text" id="new-task-title" placeholder="Task Title">
+        <textarea id="new-task-description" placeholder="Task Description"></textarea>
+        <div class="tags-container">
+            <div class="tag">Tags</div>
+            <div class="tag">Tags</div>
+            <div class="tag">Tags</div>
+            <button class="add-tag-button">+</button>
+        </div>
+        <input type="date" id="new-task-date">
+        <input type="time" id="new-task-time">
+        <div class="popup-buttons">
+            <button class="popup-button-confirm" id="add-task-confirm">Add Task</button>
+            <button class="popup-button-cancel" id="add-task-cancel">Cancel</button>
+        </div>
+    </div>
+`   );
+
+    // Event listener for opening the add task popup
+    addTaskButton.addEventListener('click', () => {
+        addTaskPopup.style.display = 'block';
+        overlay.style.display = 'block';
+    });
+
+    // Event listener for closing the add task popup
+    addTaskCancel.addEventListener('click', () => {
+        addTaskPopup.style.display = 'none';
+        overlay.style.display = 'none';
+    });
+
+    // Event listener for adding a new task
+    addTaskConfirm.addEventListener('click', () => {
```

Comment:
some bug occurs where this event listener is applied twice (or more times), causing multiple instances of tasks to show up on the calendar, so we need to sort that out.